owncloudApp/src/main/java/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -999,71 +881,92 @@ private void checkBasicAuthorization() {
+                Uri.parse(mServerInfo.mBaseUrl + "/" + getString(R.string.oauth2_url_endpoint_access)) // token endpoint
+        );
 
-        OAuth2RequestBuilder builder = oAuth2Provider.getOperationBuilder();
-        builder.setGrantType(OAuth2GrantType.AUTHORIZATION_CODE);
-        builder.setRequest(OAuth2RequestBuilder.OAuthRequest.GET_AUTHORIZATION_CODE);
+        String clientId = getString(R.string.oauth2_client_id);
+        Uri redirectUri = Uri.parse(getString(R.string.oauth2_redirect_uri));
+        AuthorizationRequest.Builder builder = new AuthorizationRequest.Builder(
+                serviceConfiguration,
+                clientId,
+                ResponseTypeValues.CODE,
+                redirectUri
+        );
 
-        if (mCustomTabPackageName != null) {
-            openUrlWithCustomTab(builder.buildUri());
-        } else {
-            openUrlInBrowser(builder.buildUri());
-        }
+        AppAuthConfiguration.Builder appAuthConfigurationBuilder = new AppAuthConfiguration.Builder();
+        appAuthConfigurationBuilder.setConnectionBuilder(new OAuthConnectionBuilder());
+        mAuthService = new AuthorizationService(this, appAuthConfigurationBuilder.build());
+
+        AuthorizationRequest request = builder.build();
+        Intent completedIntent = new Intent(this, AuthenticatorActivity.class);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, completedIntent, 0);
+
+        mAuthService.performAuthorizationRequest(request, pendingIntent);
     }
 
-    private void openUrlWithCustomTab(String url) {
-        TypedValue backgroundColor = new TypedValue();
-        getTheme().resolveAttribute(R.attr.colorPrimary, backgroundColor, true);
-
-        CustomTabsIntent intent = new CustomTabsIntent.Builder()
-                .setToolbarColor(backgroundColor.data)
-                .setStartAnimations(this, R.anim.slide_in_right, R.anim.slide_out_left)
-                .setExitAnimations(this, R.anim.slide_in_left, R.anim.slide_out_right)
-                .setShowTitle(true)
-                .build();
-
-        try {
-            intent.launchUrl(this, Uri.parse(url));
-        } catch (Exception e) {
-            Timber.e(e);
+    private void handleGetAuthorizationCodeResponse(Intent intent) {
+        AuthorizationResponse authorizationResponse = AuthorizationResponse.fromIntent(intent);
+        AuthorizationException authorizationException = AuthorizationException.fromIntent(intent);
+
+        if (authorizationResponse != null && authorizationResponse.authorizationCode != null) {
+            exchangeAuthorizationCodeForTokens(authorizationResponse);
+        } else if (authorizationException != null) {
+            updateOAuthStatusIconAndText(authorizationException);
+            showAuthStatus();
+            Timber.d("OAuth request to get authorization code failed: %s", authorizationException.error);
```

Comment:
Shouldn't we use `Timber.e` to log exceptions?

Reply:
👍